ambry-router/src/test/java/com.github.ambry.router/PutManagerTest.java

```diff
@@ -879,13 +907,16 @@ void verifyNotificationsForFailedPut() {
   private final long size;
   private long readSoFar;
   private boolean beBad = false;
+  private final boolean sendZeroSizedLastChunk;
+  private final Random random = new Random();
 
   /**
    * Constructs a MockReadableStreamChannel.
    * @param size the number of bytes that will eventually be written into this channel.
    */
-  MockReadableStreamChannel(long size) {
+  MockReadableStreamChannel(long size, boolean sendZeroSizedLastChunk) {
     this.size = size;
+    this.sendZeroSizedLastChunk = sendZeroSizedLastChunk;
```

Comment:
This variable might need a more descriptive name since it is doing more than just send zero sized last chunk.